keygen/src/keygen.rs

```diff
@@ -605,13 +639,16 @@ fn do_main(matches: &ArgMatches<'_>) -> Result<(), Box<dyn error::Error>> {
                 check_for_overwrite(outfile, matches);
             }
 
-            let keypair_name = "recover";
-            let keypair = if let Some(path) = matches.value_of("prompt_signer") {
-                keypair_from_path(matches, path, keypair_name, true)?
-            } else {
-                let skip_validation = matches.is_present(SKIP_SEED_PHRASE_VALIDATION_ARG.name);
-                keypair_from_seed_phrase(keypair_name, skip_validation, true, None, true)?
-            };
+            let derivation_path = get_derivation_path_from_matches(matches);
+            let legacy = derivation_path == None;
+
+            let keypair = keypair_from_seed_phrase(
+                "recover",
+                matches.is_present(SKIP_SEED_PHRASE_VALIDATION_ARG.name),
+                true,
+                derivation_path,
+                legacy,
+            )?;
```

Comment:
With this changes `solana_clap_utils::keypair::keypair_from_path()` is not used anymore anywhere.

Reply:
And the same for v3 now